components/raftstore/src/store/fsm/apply.rs

```diff
@@ -909,6 +909,9 @@ where
     /// in same Ready should be applied failed.
     pending_remove: bool,
 
+    /// Indicates whether the peer is unavailable. See more in `Peer`.
+    unavailable: Arc<AtomicBool>,
```

Comment:
return the state by ExecRes instead of a shared value

Reply:
Here I have a question, is `return the state by ExecRes` safe enough? For example, if a learner read occurs. I set `unavailable` to `true` before set `exist_peer's witness` to `false`. So it can Forbid reads.

Reply:
The PeerFsm is held by one raftstore thread at one time. So it changes `unavailable = true` and `is_witness=false` atomically. Reads are forbidden both before and after.

Reply:
Got it, thanks!